spec/models/post_spec.rb

```diff
@@ -0,0 +1,197 @@
+require 'rails_helper'
+require 'spec_helper'
+
+RSpec.describe Post, type: :model do
+  let!(:post) {FactoryBot.build :post}
+  let!(:post_fail) {FactoryBot.build :post, name: ""}
```

Comment:
let vs let! khác gì nhau ko bạn

Reply:
khi khai báo let thì nó chỉ có giá trị trong mỗi block đó.
Khi khai báo let! thì nó có giá trị trong mọi block it của nó. Bản chất của nó là một khối before ngầm nên nó sẽ được tính toán lại trước mỗi block it.

Reply:
chưa đủ, e nói rõ về phần này nhé

Reply:
- Có nghĩa là khai báo let chỉ có giá trị trong từng example (trong Rspec thì được đóng gói trong khối it) và nếu như bạn gọi đến một biến let nhiều lần trong cùng một example thì nó sẽ chỉ khai báo khối let một lần duy nhất thôi (ở lần gọi đầu tiên) Đây là khái niệm lazy evaluation có nghĩa là let chỉ chạy khi và chỉ khi nào được gọi đến. Mặc dù let là để định nghĩa một biến, tuy nhiên nó lại gần giống với việc gọi một method hơn.
- let! được gọi trong một khối before ngầm, do đó kết quả đã được tính toán và cache lại trước khi khổi it được thực thi. let! hữu ích khi bạn cần định nghĩa những trạng thái trước khi khối it được thực thi.